MapboxCoreNavigation/RouteController.swift

```diff
@@ -236,20 +236,30 @@ extension RouteController: CLLocationManagerDelegate {
             if !isCloseToCurrentStep && isCloseToUpComingStep {
                 
                 // Increment the step
-                routeProgress.currentLegProgress.stepIndex += 1
                 
                 // and reset the alert level since we're on the next step
                 let userSnapToStepDistanceFromManeuver = distance(along: routeProgress.currentLegProgress.currentStep.coordinates!, from: location.coordinate)
                 let secondsToEndOfStep = userSnapToStepDistanceFromManeuver / location.speed
-                incrementRouteProgressAlertLevel(secondsToEndOfStep <= RouteControllerMediumAlertInterval ? .medium : .low, location: location)
+                incrementRouteProgress(secondsToEndOfStep <= RouteControllerMediumAlertInterval ? .medium : .low, location: location, updateStepIndex: true)
                 return true
             }
         }
         
         return isCloseToCurrentStep
     }
     
-    func incrementRouteProgressAlertLevel(_ newlyCalculatedAlertLevel: AlertLevel, location: CLLocation) {
+    func incrementRouteProgress(_ newlyCalculatedAlertLevel: AlertLevel, location: CLLocation, updateStepIndex: Bool) {
+        
+        if updateStepIndex {
+            routeProgress.currentLegProgress.stepIndex += 1
+        }
+        
+        // If the step is not being updated, don't accept a lower alert level.
+        // A lower alert level can only occur when the user begins the next step.
+        if newlyCalculatedAlertLevel.rawValue < routeProgress.currentLegProgress.alertUserLevel.rawValue, !updateStepIndex {
```

Comment:
`if foo { return }` is equivalent to `guard !foo else { return }`. The only catch is that `if let foo = bar { return }` is the opposite of `guard let foo = bar else { return }`, so `if let foo = bar, baz { return }` isn’t equivalent to `guard let foo = bar, !baz else { return }`.

The `,` operator is equivalent to `&&` except with lower precedence than `let`: `if foo, bar` is equivalent to `if foo && bar`, but `if let foo = bar, baz` isn’t equivalent to `if let foo = bar && baz`.

If you want to convert this statement to a `guard` statement, first replace the `,` with `&&`, then apply [De Morgan’s](https://en.wikipedia.org/wiki/De_Morgan's_laws).